### PR TITLE
EXPERIMENTAL: Client MouseMove signal

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

WARNING: This could have serious adverse effects on performance.

Adds a signal to client mousemove as part of a test.

Applies this to the newly added MMG.

Test merge for performance metrics.

### CODE REVIEW

File: modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm [hunk 1]
```diff
@@ -223,14 +225,34 @@
...
 
 	user_to_buckle.client?.view_size.setTo(view_range)
 	user_to_buckle.pixel_y = 14
 
+/obj/machinery/mounted_machine_gun/proc/update_target_drag(client/shooting_client, atom/src_object, atom/over_object, turf/src_location, turf/over_location, src_control, over_control, params)
+	if(!istype(over_object))
+		return
+	if(istype(over_object, /atom/movable/screen))
+		return
+	to_chat(world, "new target: [over_object]")
```
Comment (Iamgoofball) [inline modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm]: you forgot to remove these

File: modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm [hunk 2]
```diff
@@ -223,14 +225,34 @@
...
+	last_target_atom = WEAKREF(over_object)
+
+/obj/machinery/mounted_machine_gun/proc/update_target(client/shooting_client, atom/new_target, location, control, params)
+	if(!istype(new_target))
+		return
+	if(istype(new_target, /atom/movable/screen))
+		return
+	to_chat(world, "new target: [new_target]")
```
Comment (Iamgoofball) [inline modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm]: ditto

File: modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm [hunk 3]
```diff
@@ -271,7 +293,8 @@
 		return FALSE
 	if(!shooting_client)
 		return FALSE
-	var/atom/target_atom = shooting_client?.mouse_object_ref?.resolve()
+	var/atom/target_atom = last_target_atom?.resolve()
+	to_chat(world, "fire_at target: [target_atom]")
```
Comment (Iamgoofball) [inline modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm]: ditto 3: the search for more money

File: modular_skyrat/master_files/code/_onclick/mousemove.dm [hunk 4]
```diff
@@ -0,0 +1,2 @@
+/client/MouseMove(object, location, control, params)
+	SEND_SIGNAL(src, COMSIG_CLIENT_MOUSEMOVE, object, location, control, params)
```
Comment (Iamgoofball) [inline modular_skyrat/master_files/code/_onclick/mousemove.dm:2]: remember to ..()

Comment (Gandalf2k15) [reply modular_skyrat/master_files/code/_onclick/mousemove.dm:2]: ya

**Comment by Gandalf2k15** [comment]
Not needed!

**Pull request closed without merging**